screen-orientation/browsing-contexts.html

```diff
@@ -0,0 +1,93 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+
+<script>
+  function loadIframe(src = "about:blank") {
+    const iframe = document.createElement("iframe");
+    iframe.src = src;
+    document.body.appendChild(iframe);
+    return new Promise(r => {
+      if (iframe.contentDocument.readyState === "complete") {
+        return r(iframe);
+      }
+      iframe.onload = () => r(iframe);
+    });
+  }
+
+  promise_test(async t => {
+    const iframe = await loadIframe();
+    await test_driver.bless("request full screen", () => {
+      return document.documentElement.requestFullscreen();
+    });
+    const { type: currentOrientation } = screen.orientation.type;
+    const isPortrait = currentOrientation.includes("portrait");
+    const newOrientation = `${isPortrait ? "landscape" : "portrait"}`;
+    const promiseToLockIFrame = iframe.contentWindow.screen.orientation.lock(
+      `${newOrientation}-secondary`
+    );
+    await screen.orientation.lock(`${newOrientation}-primary`);
```

Comment:
here we want to again capture the promise `const p = screen.orientation.lock(`${newOrientation}-primary`);`...

The `await p` before calling `unlock()`...